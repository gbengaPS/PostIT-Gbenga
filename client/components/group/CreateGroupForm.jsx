import React from 'react';
import PropTypes from 'prop-types';
import AppNav from '../common/AppNav';

/**
 * @param {object} props
 * @returns {object} -returns react element
 */
const CreateGroupForm = props => (
  (
    <div className="row">
      <AppNav />
      <div className="col s12 m6 component-container">
        <div className="modal-content">
          <form onSubmit={props.handleSubmit}>
          <p className="postit-text red-text center"> &nbsp; {props.error} </p>
          <div className="input-field">
            <input type="text" name="groupName" id="groupName"
            onChange={props.handleChange} required/>
            <label htmlFor="groupName" >Group Name</label>
          </div>
          <div className="input-field">
            <input type="text" name="groupDescription" id="groupDescription"
            onChange={props.handleChange} required/>
            <label htmlFor="password">Group Description</label>
          </div>
          <p className="row"> <input type="submit" value="Create Group"
          className="btn light-blue darken-4 col s8 offset-s2" />
          </p>
          </form>
          <p> &nbsp; </p>
        </div>
      </div>
    </div>
  )
);
CreateGroupForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  error: PropTypes.string,
};
export default CreateGroupForm;
