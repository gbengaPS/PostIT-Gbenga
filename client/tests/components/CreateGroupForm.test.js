import React from 'react';
import CreateGroupForm from '../../components/group/CreateGroupForm';

describe('CreateGroupForm component ', () => {
  const props = {
    handleChange: () => {},
    handleSubmit: () => {},
  };
  it('should render correctly', () => {
    expect(shallow(<CreateGroupForm {...props}/>)).toMatchSnapshot();
  });
});
