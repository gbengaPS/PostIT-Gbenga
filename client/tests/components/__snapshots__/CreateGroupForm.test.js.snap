// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroupForm component  should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateGroupForm
    handleChange={[Function]}
    handleSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AppNav />,
        <div
          className="col s12 m6 component-container"
        >
          <div
            className="modal-content"
          >
            <form
              onSubmit={[Function]}
            >
              <p
                className="postit-text red-text center"
              >
                   
                 
              </p>
              <div
                className="input-field"
              >
                <input
                  id="groupName"
                  name="groupName"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
                <label
                  htmlFor="groupName"
                >
                  Group Name
                </label>
              </div>
              <div
                className="input-field"
              >
                <input
                  id="groupDescription"
                  name="groupDescription"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
                <label
                  htmlFor="password"
                >
                  Group Description
                </label>
              </div>
              <p
                className="row"
              >
                 
                <input
                  className="btn light-blue darken-4 col s8 offset-s2"
                  type="submit"
                  value="Create Group"
                />
              </p>
            </form>
            <p>
                 
            </p>
          </div>
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-content"
          >
            <form
              onSubmit={[Function]}
            >
              <p
                className="postit-text red-text center"
              >
                   
                 
              </p>
              <div
                className="input-field"
              >
                <input
                  id="groupName"
                  name="groupName"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
                <label
                  htmlFor="groupName"
                >
                  Group Name
                </label>
              </div>
              <div
                className="input-field"
              >
                <input
                  id="groupDescription"
                  name="groupDescription"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
                <label
                  htmlFor="password"
                >
                  Group Description
                </label>
              </div>
              <p
                className="row"
              >
                 
                <input
                  className="btn light-blue darken-4 col s8 offset-s2"
                  type="submit"
                  value="Create Group"
                />
              </p>
            </form>
            <p>
                 
            </p>
          </div>,
          "className": "col s12 m6 component-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form
                onSubmit={[Function]}
              >
                <p
                  className="postit-text red-text center"
                >
                     
                   
                </p>
                <div
                  className="input-field"
                >
                  <input
                    id="groupName"
                    name="groupName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                  />
                  <label
                    htmlFor="groupName"
                  >
                    Group Name
                  </label>
                </div>
                <div
                  className="input-field"
                >
                  <input
                    id="groupDescription"
                    name="groupDescription"
                    onChange={[Function]}
                    required={true}
                    type="text"
                  />
                  <label
                    htmlFor="password"
                  >
                    Group Description
                  </label>
                </div>
                <p
                  className="row"
                >
                   
                  <input
                    className="btn light-blue darken-4 col s8 offset-s2"
                    type="submit"
                    value="Create Group"
                  />
                </p>
              </form>,
              <p>
                   
              </p>,
            ],
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="postit-text red-text center"
                  >
                       
                     
                  </p>,
                  <div
                    className="input-field"
                  >
                    <input
                      id="groupName"
                      name="groupName"
                      onChange={[Function]}
                      required={true}
                      type="text"
                    />
                    <label
                      htmlFor="groupName"
                    >
                      Group Name
                    </label>
                  </div>,
                  <div
                    className="input-field"
                  >
                    <input
                      id="groupDescription"
                      name="groupDescription"
                      onChange={[Function]}
                      required={true}
                      type="text"
                    />
                    <label
                      htmlFor="password"
                    >
                      Group Description
                    </label>
                  </div>,
                  <p
                    className="row"
                  >
                     
                    <input
                      className="btn light-blue darken-4 col s8 offset-s2"
                      type="submit"
                      value="Create Group"
                    />
                  </p>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "   ",
                      undefined,
                      " ",
                    ],
                    "className": "postit-text red-text center",
                  },
                  "ref": null,
                  "rendered": Array [
                    "   ",
                    undefined,
                    " ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        id="groupName"
                        name="groupName"
                        onChange={[Function]}
                        required={true}
                        type="text"
                      />,
                      <label
                        htmlFor="groupName"
                      >
                        Group Name
                      </label>,
                    ],
                    "className": "input-field",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "groupName",
                        "name": "groupName",
                        "onChange": [Function],
                        "required": true,
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Group Name",
                        "htmlFor": "groupName",
                      },
                      "ref": null,
                      "rendered": "Group Name",
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        id="groupDescription"
                        name="groupDescription"
                        onChange={[Function]}
                        required={true}
                        type="text"
                      />,
                      <label
                        htmlFor="password"
                      >
                        Group Description
                      </label>,
                    ],
                    "className": "input-field",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "groupDescription",
                        "name": "groupDescription",
                        "onChange": [Function],
                        "required": true,
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Group Description",
                        "htmlFor": "password",
                      },
                      "ref": null,
                      "rendered": "Group Description",
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <input
                        className="btn light-blue darken-4 col s8 offset-s2"
                        type="submit"
                        value="Create Group"
                      />,
                    ],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "btn light-blue darken-4 col s8 offset-s2",
                        "type": "submit",
                        "value": "Create Group",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "   ",
              },
              "ref": null,
              "rendered": "   ",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppNav />,
          <div
            className="col s12 m6 component-container"
          >
            <div
              className="modal-content"
            >
              <form
                onSubmit={[Function]}
              >
                <p
                  className="postit-text red-text center"
                >
                     
                   
                </p>
                <div
                  className="input-field"
                >
                  <input
                    id="groupName"
                    name="groupName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                  />
                  <label
                    htmlFor="groupName"
                  >
                    Group Name
                  </label>
                </div>
                <div
                  className="input-field"
                >
                  <input
                    id="groupDescription"
                    name="groupDescription"
                    onChange={[Function]}
                    required={true}
                    type="text"
                  />
                  <label
                    htmlFor="password"
                  >
                    Group Description
                  </label>
                </div>
                <p
                  className="row"
                >
                   
                  <input
                    className="btn light-blue darken-4 col s8 offset-s2"
                    type="submit"
                    value="Create Group"
                  />
                </p>
              </form>
              <p>
                   
              </p>
            </div>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="modal-content"
            >
              <form
                onSubmit={[Function]}
              >
                <p
                  className="postit-text red-text center"
                >
                     
                   
                </p>
                <div
                  className="input-field"
                >
                  <input
                    id="groupName"
                    name="groupName"
                    onChange={[Function]}
                    required={true}
                    type="text"
                  />
                  <label
                    htmlFor="groupName"
                  >
                    Group Name
                  </label>
                </div>
                <div
                  className="input-field"
                >
                  <input
                    id="groupDescription"
                    name="groupDescription"
                    onChange={[Function]}
                    required={true}
                    type="text"
                  />
                  <label
                    htmlFor="password"
                  >
                    Group Description
                  </label>
                </div>
                <p
                  className="row"
                >
                   
                  <input
                    className="btn light-blue darken-4 col s8 offset-s2"
                    type="submit"
                    value="Create Group"
                  />
                </p>
              </form>
              <p>
                   
              </p>
            </div>,
            "className": "col s12 m6 component-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  onSubmit={[Function]}
                >
                  <p
                    className="postit-text red-text center"
                  >
                       
                     
                  </p>
                  <div
                    className="input-field"
                  >
                    <input
                      id="groupName"
                      name="groupName"
                      onChange={[Function]}
                      required={true}
                      type="text"
                    />
                    <label
                      htmlFor="groupName"
                    >
                      Group Name
                    </label>
                  </div>
                  <div
                    className="input-field"
                  >
                    <input
                      id="groupDescription"
                      name="groupDescription"
                      onChange={[Function]}
                      required={true}
                      type="text"
                    />
                    <label
                      htmlFor="password"
                    >
                      Group Description
                    </label>
                  </div>
                  <p
                    className="row"
                  >
                     
                    <input
                      className="btn light-blue darken-4 col s8 offset-s2"
                      type="submit"
                      value="Create Group"
                    />
                  </p>
                </form>,
                <p>
                     
                </p>,
              ],
              "className": "modal-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="postit-text red-text center"
                    >
                         
                       
                    </p>,
                    <div
                      className="input-field"
                    >
                      <input
                        id="groupName"
                        name="groupName"
                        onChange={[Function]}
                        required={true}
                        type="text"
                      />
                      <label
                        htmlFor="groupName"
                      >
                        Group Name
                      </label>
                    </div>,
                    <div
                      className="input-field"
                    >
                      <input
                        id="groupDescription"
                        name="groupDescription"
                        onChange={[Function]}
                        required={true}
                        type="text"
                      />
                      <label
                        htmlFor="password"
                      >
                        Group Description
                      </label>
                    </div>,
                    <p
                      className="row"
                    >
                       
                      <input
                        className="btn light-blue darken-4 col s8 offset-s2"
                        type="submit"
                        value="Create Group"
                      />
                    </p>,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "   ",
                        undefined,
                        " ",
                      ],
                      "className": "postit-text red-text center",
                    },
                    "ref": null,
                    "rendered": Array [
                      "   ",
                      undefined,
                      " ",
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          id="groupName"
                          name="groupName"
                          onChange={[Function]}
                          required={true}
                          type="text"
                        />,
                        <label
                          htmlFor="groupName"
                        >
                          Group Name
                        </label>,
                      ],
                      "className": "input-field",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "id": "groupName",
                          "name": "groupName",
                          "onChange": [Function],
                          "required": true,
                          "type": "text",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Group Name",
                          "htmlFor": "groupName",
                        },
                        "ref": null,
                        "rendered": "Group Name",
                        "type": "label",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          id="groupDescription"
                          name="groupDescription"
                          onChange={[Function]}
                          required={true}
                          type="text"
                        />,
                        <label
                          htmlFor="password"
                        >
                          Group Description
                        </label>,
                      ],
                      "className": "input-field",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "id": "groupDescription",
                          "name": "groupDescription",
                          "onChange": [Function],
                          "required": true,
                          "type": "text",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Group Description",
                          "htmlFor": "password",
                        },
                        "ref": null,
                        "rendered": "Group Description",
                        "type": "label",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <input
                          className="btn light-blue darken-4 col s8 offset-s2"
                          type="submit"
                          value="Create Group"
                        />,
                      ],
                      "className": "row",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "btn light-blue darken-4 col s8 offset-s2",
                          "type": "submit",
                          "value": "Create Group",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                    ],
                    "type": "p",
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "   ",
                },
                "ref": null,
                "rendered": "   ",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
